tests/test_temporal.py

```diff
@@ -2356,3 +2377,754 @@ async def test_beta_graph_parallel_execution_in_workflow(client: Client):
+        # Test selecting the second model
+        output = await client.execute_workflow(
+            MultiModelWorkflowSelectModel2String.run,
+            args=['Hello'],
+            id='MultiModelWorkflowSelectByInstance',
+            task_queue=TASK_QUEUE,
+        )
+        assert output == 'Second model response'
+
+
+async def test_temporal_agent_multi_model_backward_compatibility():
+    """Test that single-model agents work as before (backward compatibility)."""
+    test_model = TestModel()
+
+    agent = Agent(test_model, name='backward_compat_test')
+    temporal_agent = TemporalAgent(agent, name='backward_compat_test')
+
+    # Should only have default model registered
+    assert list(temporal_agent._registered_model_instances.keys()) == ['default']  # pyright: ignore[reportPrivateUsage]
+
+    # Should work without additional_models parameter
+    assert isinstance(temporal_agent.model, TemporalModel)
+    assert temporal_agent.model.wrapped == test_model
+
+    # Activity names should only include the default request/stream pair
+    activity_names = {
+        name
+        for activity in temporal_agent.temporal_activities
+        if (name := _temporal_activity_name(activity)) is not None
+    }
+    assert 'agent__backward_compat_test__model_request' in activity_names
+    assert 'agent__backward_compat_test__model_request_stream' in activity_names
+
+
+async def test_temporal_agent_multi_model_outside_workflow():
+    """Test that multi-model agents work outside workflows (using wrapped agent behavior)."""
+    test_model1 = TestModel(custom_output_text='Model 1 response')
+    test_model2 = TestModel(custom_output_text='Model 2 response')
+
+    agent = Agent(test_model1, name='outside_workflow_test')
+    temporal_agent = TemporalAgent(
+        agent,
+        name='outside_workflow_test',
+        additional_models={'secondary': test_model2},
+    )
+
+    # Outside workflow, should use default model from wrapped agent
+    result = await temporal_agent.run('Hello')
+    assert result.output == 'Model 1 response'
+
+    # Outside workflow, the temporal agent still uses the wrapped model
+    # Model selection via additional_models only works inside workflows
+    # This test verifies that the agent works correctly outside workflows
+    # even when additional models are registered
+    result = await temporal_agent.run('Hello')
+    assert result.output == 'Model 1 response'
+
+
+async def test_temporal_agent_select_model_string_passthrough():
+    """Test that unregistered model strings pass through to infer_model."""
+    test_model1 = TestModel()
+
+    agent = Agent(test_model1, name='string_passthrough_test')
+    temporal_agent = TemporalAgent(
+        agent,
+        name='string_passthrough_test',
+    )
+
+    # Unregistered model string should pass through directly
+    selected = temporal_agent._select_model('openai:gpt-4o')  # pyright: ignore[reportPrivateUsage]
+    assert selected == 'openai:gpt-4o'
```

Comment:
I don't want to test private methods like this, just the behavior of the feature.

A lot of the tests here seem unnecessary.